Hard code known true contiguity settings for unbacked SymInts

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #95003

Extracted from https://github.com/pytorch/pytorch/pull/94523 which has E2E test

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

ezyang committed:
Hard code known true contiguity settings for unbacked SymInts

Extracted from https://github.com/pytorch/pytorch/pull/94523 which has E2E test

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

voznesenskym commented on c10/core/TensorImpl.cpp:
```diff
@@ -1212,6 +1212,27 @@ void TensorImpl::empty_tensor_restride_symint(MemoryFormat memory_format) {
   refresh_contiguous();
+  // hard code some known true settings, for unbacked case
+  // TODO: avoid chundering into the guards for computing these
+  switch (memory_format) {
```
if you don't set true, do they default false?

ezyang replied:
no, they would be set to some complicated expression from the guard-less bool implementation that would evaluate to true (often, sympy cannot figure it out though)

ezyang committed:
Update on "Hard code known true contiguity settings for unbacked SymInts"


Extracted from https://github.com/pytorch/pytorch/pull/94523 which has E2E test

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge
